pipeline/src/node_properties/model_inference_properties.rs

```diff
@@ -39,3 +43,13 @@ pub struct ModelInferenceRuntime {
     /// Map key is pipeline node id, val is unique id assigned to it at runtime.
     pub infer_node_unique_ids: Option<BTreeMap<String, i32>>,
 }
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct PerClassThreshold {
+    /// Use a negative class_id value (i.e: -1) to define a global threshold for all the classes.
+    ///
+    /// To define individual class threshold, we accept integers starting from 0 using the same
+    /// order as the model class labels.
+    pub class_id: i32,
```

Comment:
@SIGLOST Thinking out loud - wondering if it would be more scalable if we used class labels instead of id's here. Here's the rationale 
- for nvinfer, the thresholds are specified using class id's, but some other inference element (ex https://github.com/openvinotoolkit/dlstreamer_gst/wiki/gvainference ) may require differently ?
- should console / API have to know how to map labels to class id's ? if we specify this using class labels, then this mapping can live in lumeod and api/console can deal with the high level specification using labels.

Reply:
So if the inference node in the future might accept both nvinfer (that takes "class_id") and other inference element (that takes "class_label") I think the more flexible option for the API would be expand this struct to have both "class_id" and "class_label".

The "class_id" is an integer starting from 0 that uses the same order as the classes are defined so should not be difficult to populate on the root element that retrieves this info. 

That way console and API users would easily fill the required fields, depending on the circumstances. and backend components would not be required to read and parse additional files (maybe multiple times).

If that means a lot of work or we have some blocker, ofc we can switch here to just "class_label" and propagate the mapping for the backend components

Reply:
> The "class_id" is an integer starting from 0 that uses the same order as the classes are defined so should not be difficult to populate on the root element that retrieves this info.

This is an implicit assumption specific to nvidia's implementation - so IMHO it is likely a wise thing to leave it out. From a user standpoint, and from a cloud standpoint, labels are the ones that really matter.

Reply:
Yes that's true the end-user (at console level) just cares about the "labels", but the usage of "class_id" (or an integer class) is widely used and attached to the inference engines layer, as with them is always possible to make a 1:1 relation with the neural network output. This approach is used mainly because depending on the implementation you can map the int output to different needs or languages (english, chinese, etc). In fact nvinfer takes a smart approach as it opens the possibility of each user customize the classes names they want to use, without causing any impact on the configuration.

For example in tensorflow you have some utils to later associate of the integer classes with labels (basically a list of strings following the same order): 
https://www.tensorflow.org/datasets/api_docs/python/tfds/features/ClassLabel

Reply:
on Openvino:
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/samples/object_detection_sample_ssd/main.cpp#L323

I didn't found any high-level config on this framework related with per-class config on the gva component